### DISCUSSION TITLE
Forcing the optimizer to consider linear plans only?

### DISCUSSION THREAD

**Discussion by remysucre** [duckdb/duckdb#3935] [Q&A] [answered]
Is there a way to force the optimizer to consider linear plans (including zig zag plans) only? I'm working on a join algorithm that only works for linear plans, and I'm looking to piggy back on DuckDB's query planner.

**Comment by Mytherin** [comment, marked as answer]
Not out of the box. I suggest you have a look at the join order optimizer.
